sympy/polys/polytools.py

```diff
@@ -5622,12 +5622,12 @@ def _sorted_factors(factors, method):
         def key(obj):
             poly, exp = obj
             rep = poly.rep.rep
-            return (exp, len(rep), rep)
+            return (exp, len(rep), len(poly.gens), rep)
```

Comment:
I added that test. This is ready to go. I'll commit in 24h if there is no objection.